content/data/tanzu-gemfire/blog/geode-distributed-sequences.md

```diff
@@ -0,0 +1,92 @@
+
+## Why?
+In order to move a step closer to cloud-native applications and a little away from legacy RDBMS, it would be nice to have distributed sequences/counters directly managed by [Apache Geode](https://geode.apache.org/).
+
+## How?
+The first thing we have to keep in mind is that, even though we can generate unique sequential distributed sequences with Geode, **we have to loosen the restriction about having no-gaps between the sequences.** If we don’t, there will always be a potential contention point (either while computing sequences or while sequentially executing the application work using the sequences) and, thus, fully parallel processing can not be truly achieved.
+If you are willing to allow gaps, continue reading further, if you don’t… well, this tool is not for you, don’t waste any more of your time :-).
+
+## The Solution
+Instead of embedding the sequence generation within the real-time processing, use the [Geode Sequence Generator](https://github.com/jujoramos/geode-sequence-generator) and invoke the service occasionally to retrieve a big enough set of counters to work on.
+Ideally, the service should be invoked during client startup and the batch requested should be big enough to avoid multiple calls to the service in real-time. The above alleviates the contention on the servers and maintains scalability, allowing several client applications to truly process batches of work in parallel using the set of assigned sequentially generated IDs.
+
+##How Does it Work?
+The tool primarily relies on `PERSISTENT` regions, the `Distributed Lock Service` and `Function Execution`.
+There’s only one `Region` acting as the backing store, no matter how many different sequences are managed, and a `Distributed Lock` is acquired every time a sequence needs to be computed (thus the recommendation to avoid invoking the service multiple times in real-time and get a big enough batch of sequences to work on from the start).
+To initialize the backing `Region`, the tool executes an initialization `Function` through the usage of `FunctionService.onServers`, allowing the user to choose exactly which servers will be used as the back end for the sequence generation (all by default).
+To generate or retrieve a batch of sequences, the tool executes a `Function` through `FunctionService.onRegion`, guaranteeing that only the server hosting the primary copy of the data will be reached.
+
+## Examples?
+All right, time to write some code, finally!.
+
+### Compile and Deploy
+The first step is to download the tool, build it and deploy it to a running Apache Geode cluster. We list the functions at the very end just to verify that they have been correctly registered through the gfsh deploy command.
+
+![img](/images/data-blogs/tanzu-gemfire/geode-distributed-sequences/geode-distributed-sequences-gfsh-deploy.png)
```

Comment:
In the future, we prefer any commands that a user is intended to run be formatted as code blocks as opposed to screen shot images. If you want to include command output as a means of verification, that can also be a code block or an image, but preferably is limited to one command at a time.

Reply:
Yes, completely agree.  Moving forward this is something we will continue to communicate throughout the team.